src/client.ts

```diff
@@ -4991,21 +4990,22 @@ export class MatrixClient extends EventEmitter {
     // XXX: Intended private, used in code.
     public createMessagesRequest(
         roomId: string,
-        fromToken: string,
-        limit: number,
+        fromToken: string | null,
```

Comment:
This is relying on unspecced behaviour of Synapse. A `from` token is required by the spec.

https://spec.matrix.org/v1.1/client-server-api/#get_matrixclientv3roomsroomidmessages

Reply:
Thanks, I did realise that afterwards. It seems that synapse supports that because the android client used to make some calls without that parameter.
Me and @clokep are discussing how EW could get its hands on a pagination token without going through the whole sync flow as it won't fit the bill here.

Won't merge this pull request until we have solved this

Reply:
To cross-reference, see matrix-org/synapse#5538. There's desire to remove this capability from Synapse (in order to be spec compliant).

It was suggested that there should be a sync token somewhere to use in order to use this API, but I think @gsouquet wants to not have to sync the entire filtered request, but simply paginate through a filtered set of messages, which seems like quite a reasonable thing to do?

Reply:
it's very reasonable in my opinion to have `from` be optional, largely for cases like this. It just needs to be in the spec as such, or at least as an MSC with net-positive feedback.

Reply:
This has been proposed in https://github.com/matrix-org/matrix-doc/pull/3567
I will consider this pull request unblocked as it's likely to be approved and enter FCP very soon